### DISCUSSION TITLE
Option to provide default when license is `NOASSERTION`

### DISCUSSION THREAD

**Discussion by spenserblack** [lowlighter/metrics#422] [Ideas]
For example, a lot of Rust projects [(including Rust itself)](https://github.com/rust-lang/rust) are dual-licensed, which results in `NOASSERTION`, even though the repository *is* licensed.

By providing a default license value (possibly per-language), a user can set a value that makes more sense.

I think it could be something like this:
```yaml
licenses_default: 'rust:MIT OR Apache-2.0,c:GPL-3.0' # etc...
```

There is a danger that this can result in inaccurate information if a user isn't careful with this option, though.

**Comment by lowlighter** [comment]
In #400 there was a fix to display `Other` instead of `NOASSERTION` but after re-reading it, I don't think it patched all occurences of licenses display. Maybe just patching the remaining occurence to display `Other` would be sufficient ?

I can see `licenses_default` being practical for your own repositories, but in this case I would be tempted to make it per repository rather than per languages (which seems really broad and like you said would be inaccurate and misleading if not handled well)

**Reply by spenserblack** [reply]
>I don't think it patched all occurences of licenses display

Yeah, I'm using `v3.11` and the base metrics still report `NOASSERTION`.

>per repository rather than per languages (which seems really broad and like you said would be inaccurate and misleading if not handled well)

The reason I was thinking per-language is because the metric itself is kind of broad, at least for base metrics. It reports *prefers*, not an exact count. I'm not sure how common my situation is, but, at least personally, I can be confident that almost all of my Rust projects have the same license.

I feel like license preference is often more per-ecosystem than per-repo. That is, I think most people usually stick with the same license for their cargo packages, same license for their npm packages, etc.

But I'm by no means 100% sure about this. I just wanted to bring up the discussion and see if it was reasonable 🙂

**Reply by lowlighter** [reply]
I don't think it'll be implemented. I can't remember when but there was a change to replace the NOASSERTION text into "No license preference", it's probably good enough

The addition of a new option for such a edge case is probably overkill 😅

**Reply by spenserblack** [reply]
Agreed 🙂
